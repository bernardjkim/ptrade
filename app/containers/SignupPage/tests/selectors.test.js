import { fromJS } from 'immutable';
import {
  selectSignupPageDomain,
  makeSelectEmail,
  makeSelectError,
  makeSelectLoading,
  makeSelectPassword,
  makeSelectUsername,
} from '../selectors';

const mockedState = fromJS({
  signupPage: {
    username: 'username',
    email: 'test@email.com',
    password: 'password',
    loading: false,
    error: 404,
  },
});

describe('selectSignupPageDomain', () => {
  it('should select the SignupPage state', () => {
    expect(selectSignupPageDomain(mockedState)).toMatchSnapshot();
  });
});

describe('makeSelectUsername', () => {
  const usernameSelector = makeSelectUsername();
  it('should select the username', () => {
    expect(usernameSelector(mockedState)).toEqual('username');
  });
});

describe('makeSelectEmail', () => {
  const emailSelector = makeSelectEmail();
  it('should select the email', () => {
    expect(emailSelector(mockedState)).toEqual('test@email.com');
  });
});

describe('makeSelectPassword', () => {
  const passwordSelector = makeSelectPassword();
  it('should select the password', () => {
    expect(passwordSelector(mockedState)).toEqual('password');
  });
});

describe('makeSelectLoading', () => {
  const loadingSelector = makeSelectLoading();
  it('should select the loading', () => {
    expect(loadingSelector(mockedState)).toEqual(false);
  });
});

describe('makeSelectError', () => {
  const errorSelector = makeSelectError();
  it('should select the error', () => {
    expect(errorSelector(mockedState)).toEqual(404);
  });
});
