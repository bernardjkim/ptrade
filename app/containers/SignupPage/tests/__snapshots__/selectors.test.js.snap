// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectSignupPageDomain should select the SignupPage state 1`] = `
Immutable.Map {
  "username": "username",
  "email": "test@email.com",
  "password": "password",
  "loading": false,
  "error": 404,
}
`;
